[Goh YC, Gabriel] iP

gycgabriel opened the pull request:
# Nekobot

> “When I try to talk to people, it’s like time has stopped and nobody answers. 
> Wait, so I guess this is what they call magic…”

Don't fret! Now this cute cat bot will respond to your task queries!

Maybe you've heard of that other task management app. But here's why you can count on Nekobot to be so much better!

* text-based
* easy to use
* cute cat greets you on startup

## Downloading Nekobot
1. Go to Releases page
2. Download the latest release
3. Open it and feel blessed :relaxed:

The best part of it: Nekobot is free!

## Features
- [x] Managing Todos, Deadlines, Events
- [x] Time-understanding app
- [ ] Reminders (coming soon!)

```
                       .,*#&@@@@@@@@@@@@@@&*                                                        
                                  .,/(%&@&&@@@@@@@@@@@@@@@@@%* .,/(%&@@@@@@@@@@@@&.                 
                                ,%@@@@@@@&&&@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@,                 
                            /@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@&&%#(##@@@@@*                 
                        .(@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@&(&&%&&&(@@@@@*                 
                     .,%@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@/&%%&(@@@@@,                 
              .(@@@@@@@@@@@@@@@@@@@@@@@&@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@(&#%@@@@&.                 
         *&@@@@@@@@@@@@@@@@@@@@@@@@@@&@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@%/@@@@@/                  
     .@@@@@@@@@@&(*((@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@&&&&%.                  
      *@@@@@&*(#&%%%#@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@/&@@@@@@@@@@@@@@@@@@@&&&,         ..        
       ,@@@@@@(%%%&#@@@@@@@@&@@&@@@@@@@@@@@&@@@@@@@@@@@@#@(@@@@@@@@@@@@@@@@@@@@.        .((((,/(*.  
        .&@@@@@@*%&(@@@@@@@@@@@@@@@@@@@@@@@/@@@@@@@@@@@%%@@@(@@@@@@@@@@@@@&@@@@.        ,((((((((/  
          ,@@@@@@@/%@@@@@@@@@@@@@@@@@@@@@@(&%@@@@@@@@@@(@@@@@@/@@@@@@@@@@@@&@@#@*        ..(((/,.   
            *@@&&&&@@@@@@@@@@@@@@@@@@%@@@(@@(@@@@@@@@@&%@@@@@@@@/@@@@@@@@@@#@@@/.%.                 
              ,&&&@@@@@&@@@@@@@@@@@@@@@@&%@@#@@@@@@@@@/@@@@@@@@@@@/@@@@@@@@#&%@@,  ..   .(*         
                .%&@@@@#@@@@@@@@@@@@@@@@#@@@&%@@@@@@@%&@@%. ....... #@@@@@&#&& &&                   
                  %@@@@#@@@@@@@@@@@(@@@#@&@@@(@@@@@%@#@@@%**%%***/*@%&@@@@&(&&  (.     .,           
                  %&@@@@#@@@@@@%@@#.....,@@@@&&@@@%##@@@@@*((((/((/@&&@@@@&(@#   .                  
                 .(*@@@@%&@@@@@, .*//*@*/*@@@@(@(/@/@@@@@@@#*((#@&@@&&@@@&/%&.                      
                 ...%&&@@(@@@@@#@,*/(((((/@@@@@@@@@@@@@@@@@@@@(/#,&(#&%@&&.&                        
                    .&&%&@%@@@@&&@*((&,/@@@@@@@@@@@@@@@@@@@@@@@/#%%&#%.%@...                        
                      *#&%&&(@@@(@&&&&&&@@@@@@@@@@@@@@@@@@@@@@@@@@@@@#/.#                           
                       .%*%&&%(&(*@&&&&&@@@@@@@@@@#%@@@@@@@@@@@@@@@&..,,..                          
                       ....(&&*(#,%#@%**/&@@@@@@@@@@@@@@@@@@@@#,./#@%#/*//(.                        
                            ,&.,,,.,,,,,,,,,*@@@@@@@@@@@@&,.,,,,,,,,,,,,,,,,.                       
                           .(***.,,,,,,,,,,,*(,... #@@%..,,,/*,,,,,,,,,,,,,..                       
                           .,,,,,,.....(*,,,./&@@...,,.@@@#,,,,,,.,*.,,,.....                       
                            ...,,,,(@%#&@@@@@@@@@*.,//(@@@@@@@@@@&@@/.,.....                        
                            ...,,,.@@@@@@@@@@@@@@/&%%(*@@@@@@@@@@@&@%#&*....                        
```


### To run using source code:
`new Duke([FILEPATH_TO_STORAGE]).run()`

Commit:
Add Gradle support

gycgabriel committed:
Add greet, echo, exit

gycgabriel committed:
Add add, list

gycgabriel committed:
Add mark as done

gycgabriel committed:
Add Todo, Event, Deadline

gycgabriel committed:
Add automated text ui testing

gycgabriel committed:
Add custom exceptions

gycgabriel committed:
Add delete

gycgabriel committed:
Add Commands enum

gycgabriel committed:
Add extra functionality

Done, Delete by task name
Disallow adding duplicate tasks
Optimize string concat with StringBuilder

gycgabriel committed:
Add cat ascii art in txt file

gycgabriel committed:
Add save

gycgabriel committed:
Add dates and times

gycgabriel committed:
Merge branch 'branch-Level-7'

gycgabriel committed:
Merge branch 'branch-Level-8'

gycgabriel committed:
Fix datetime flexibility and event instantiation

gycgabriel committed:
Refactor to use more OOP

gycgabriel committed:
Put all classes into package duke

gycgabriel committed:
Add Parser class

gycgabriel committed:
Add JUnit test for Duke and Storage

gycgabriel committed:
Generate jar file from intellij artifacts

gycgabriel committed:
Add javadocs comments

gycgabriel committed:
Add find

gycgabriel committed:
Adhere to coding standard

gycgabriel committed:
Merge branch 'branch-A-CodingStandard'

gycgabriel committed:
Merge branch 'branch-Level-9'

gycgabriel committed:
Apply braces to if statements

gycgabriel committed:
Update Readme.md

gycgabriel committed:
Merge branch 'add-gradle-support'

gycgabriel committed:
Merge branch 'master' of https://github.com/gycgabriel/ip

gycgabriel committed:
Set up Gradle, checkstyle

gycgabriel committed:
Set up javafx dependencies

gycgabriel committed:
Add GUI

gycgabriel committed:
Add greeting

gycgabriel committed:
Customize GUI

gycgabriel committed:
Fix png to be case-sensitive for jar file

Null pointer exception after building shadowJar when case of file path does not match actual path.

Jar cannot run.

Let's
* Change file extension and file path to use lowercase png so jar can run.
* Fix fxml files to use the same javafx version as build.gradle
* Fix accidental wildcard import
* Fix tests to work after GUI implementation

Preshita01 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,194 @@
+        }
+    }
+
+    private boolean parseParameters(String argument, Duke duke) throws DukeException {
```
Perhaps you want to change the name of this method to sound more like a boolean? (e.g. canParseParams)

gycgabriel replied:
I think if it was a method to return a boolean I would name it `checkIfParsedParameters` but I think this method does more than that, it parses the parameters storing them in Parser variables. Thus I don't think it is a good idea to name it as a method that merely returns a boolean value. ^^

gycgabriel replied:
I was also hoping to get feedback on this actually: 
> parses the parameters storing them in Parser class variables

Do you think it's a good idea in terms of code quality?

In terms of CS2030, encapsulation/data hiding needs Parser class as private variables and maybe a getter function for Duke class to access them. It's just that that means getters for parameters, description, at, by, .... which feels like a lot of boilerplate code. Returning a String[] means the work to find out what each String contains has to be outside the Parser.

I'm curious to know your solution to this

Preshita01 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,158 @@
+
+
+    /**
+     * Gets response to user input.
```
Should there be a '@param' line (for the parameter 'input') and a '@return' line?

gycgabriel replied:
Thanks for pointing out

Preshita01 commented on src/main/java/duke/DukeException.java:
```diff
@@ -0,0 +1,103 @@
+
+}
+
+class TaskIndexNotInteger extends DukeException {
```
Maybe you can include more descriptive header comments for the classes and public methods here? I think there are also other methods in the rest of the code for which header comments can be added.

gycgabriel replied:
Ok

Preshita01 commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,79 @@
+    }
+
+    /**
+     * Convert deadline into savable format.
```
Should this first sentence be '**Converts** deadline...' instead of 'Convert'? I think this occurs at a few other places in the code as well.

gycgabriel replied:
Thanks for pointing out 😄

Preshita01 reviewed (commented):
Neat code overall 👍 , with just some minor points for improvement. It is also great to see that you have gotten really creative with your iP!

fullfatwasabi commented on src/main/java/duke/DialogBox.java:
```diff
@@ -0,0 +1,67 @@
+ * This control represents a dialog box consisting of an ImageView to represent the speaker's face and a label
+ * containing text from the speaker.
+ * https://se-education.org/guides/tutorials/javaFx.html
+ */
```
haha don't forget to change this! :)

gycgabriel replied:
Thanks for pointing out!

fullfatwasabi commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,79 @@
+    }
+
+    /**
+     * String representation of deadline for printing.
```
Just to add to what Preshita01 mentioned about the tenses for your header comments, here should be "Returns string..." I think.

gycgabriel replied:
Ok 😅

fullfatwasabi reviewed (commented):
Love your theming haha, overall just some small things with the header comments and the boolean naming that Preshita01 mentioned! :)

gycgabriel committed:
Add load

Refactor Parser class to adhere to coding standard, close #1

gycgabriel committed:
Add auto-load from start

gycgabriel committed:
Add Hawking Time Parser

gycgabriel committed:
Rearrange import statements

gycgabriel committed:
Set default empty enter to list

gycgabriel committed:
Add assert statements for Storage and Parse

Make ParsedInput variables final

gycgabriel committed:
Modify to adhere to coding quality

gycgabriel committed:
Merge pull request #2 from gycgabriel/branch-A-Assertions

Add assert statements for Storage and Parse

gycgabriel committed:
Merge branch 'master' into branch-A-CodeQuality

gycgabriel committed:
Merge pull request #3 from gycgabriel/branch-A-CodeQuality

Branch A-CodeQuality

gycgabriel committed:
Fix creating null tasks

gycgabriel committed:
Update README.md

gycgabriel committed:
Update tests

gycgabriel committed:
Fix Duke not constructed because constructor threw errrors

When could not load file, Duke instance not created.

Move error handling from outside Duke constructor to inside Duke constructor. Add MainWindow reference to print error messages from Duke instance.

gycgabriel committed:
Add title to GUI window

gycgabriel committed:
Add transparency to images

gycgabriel committed:
Adjust image margins

gycgabriel committed:
Revert "Adjust image margins"

This reverts commit bbbe8eb0ecc1590a6316a285465c29dae608d70c.

gycgabriel committed:
Resize image

gycgabriel committed:
Delete PNG files

gycgabriel committed:
Add png files in lowercase

gycgabriel committed:
Update messages shown when loading file on start

gycgabriel committed:
Update header comments to coding standard

gycgabriel committed:
Refactor enums into appropriate classes

gycgabriel committed:
Refactor code into sub-packages

gycgabriel committed:
Refactor commands to use more OOP

getResponse method in Duke handled command behaviour and became a long method.

Let's distribute command behaviour into run method in each specific command class.

gycgabriel committed:
Fix task IOOB exception message

gycgabriel committed:
Add javadocs for Ui methods

gycgabriel committed:
Add Ui.png

gycgabriel committed:
Update README.md

gycgabriel committed:
Allow bye command when tasklist is empty
